src/PhpSessionPersistence.php

```diff
@@ -191,6 +192,10 @@ private function startSession(string $id, array $options = []) : void
      */
     private function regenerateSession() : string
     {
+        if (PHP_SESSION_ACTIVE === session_status()) {
+            session_destroy();
+        }
+
         session_write_close();
```

Comment:
This is a noop. If session exists then it is destroyed before it reaches this call

Reply:
@Xerkus what do you mean? Only session with old session id that will be destroyed. New session data with the regenerate id will be saved to new session id.

Reply:
Unless I misunderstand how it works, `session_write_close()` is a noop if session is not started. And whenever there *is* a session, conditional block above invokes  `session_destroy()`. At which point no session exists again.

Reply:
I tested this in real mezzio example application at https://github.com/samsonasik/mezzio-authentication-with-authorization/tree/issue-demo . 

In the `issue-demo` branch, I set session save path to './data' directory. I registered my forked repo at composer.json: https://github.com/samsonasik/mezzio-authentication-with-authorization/blob/a5454dedf3cb4713bd27d4165863b23ad22af89a/composer.json#L76-L82

Step to reproduce:

- git clone git@github.com:samsonasik/mezzio-authentication-with-authorization.git
- git checkout issue-demo
- follow the README.md
- open localhost:8080 and the csrf data written to file at './data/sess_*' file
- when logged in, the csrf data in the './data` directory will be removed and new session file created during session regenerate id.

Reply:
@samsonasik which relates how to `session_write_close()` never reaching condition where it performs any action?

https://github.com/php/php-src/blob/fabe6aa7dc35c10d0c034bec743631ec4a64b0d0/ext/session/session.c#L2584
```c
	if (PS(session_status) != php_session_active) {
		RETURN_FALSE;
	}
```

Reply:
what your suggestion? add `else` control after `if` ?

Reply:
it is a noop, remove it.

Reply:
ok, I've removed `session_write_close()`